app/octopus_graphql_api_client.py

```diff
@@ -0,0 +1,173 @@
+import json
+import aiohttp
+from datetime import (datetime, timedelta, time)
+from intelligent_dispatches import IntelligentDispatchItem, IntelligentDispatches
+from contextlib import suppress
+import ciso8601
+
+intelligent_dispatches_query = '''query {{
+  plannedDispatches(accountNumber: "{account_id}") {{
+    startDt
+    endDt
+    delta
+    meta {{
+      source
+      location
+    }}
+  }}
+  completedDispatches(accountNumber: "{account_id}") {{
+    startDt
+    endDt
+    delta
+    meta {{
+      source
+      location
+    }}
+  }}
+}}'''
+
+def parse_datetime(dt_str: str) -> dt.datetime | None:
```

Comment:
I am getting an error here because `dt` is not imported:
> NameError: name 'dt' is not defined

Some items from `datetime` are already imported, I think you need to remove the `dt.` qualifier and ensure all the others are imported (line 3).

Note, there seem to be some other unresolved references such as `DATETIME_RE`, `Any`, `UTC`, `DEFAULT_TIME_ZONE`, `click` that need importing.

Reply:
This isn't my code, it's imported from the Home Assistant integration for Octopus Energy, and I cannot find where any of these are defined in that code either. :/

Are you getting errors during compilation or linting?

Reply:
Yes, I can't run the application as it stands:

```
Traceback (most recent call last):
  File "octograph/app/octopus_to_influxdb.py", line 13, in <module>
    from octopus_graphql_api_client import OctopusEnergyApiClient
  File "octograph/app/octopus_graphql_api_client.py", line 29, in <module>
    def parse_datetime(dt_str: str) -> dt.datetime | None:
                                       ^^
NameError: name 'dt' is not defined

Process finished with exit code 1
```

Reply:
How are you running it? I only ever run it by building the image and running the container, which should mean everything that is needed is there;
```
docker build --tag octograph:dq .
docker run --rm --name octograph --network octograph -v "/home/dq/octograph.home.ini":/etc/octograph.ini:ro -e OCTOGRAPH_OCTOPUS_API_KEY="sk_live_blah" -e OCTOGRAPH_INFLUXDB_TOKEN="blah" octograph:dq --config-file=/etc/octograph.ini
```
I've updated the code in a new commit but this works for me both pre and post changes.

Reply:
I am able to run this now after your latest commit, but only because my config doesn't use `OctopusGraphQlApiClient`. This class still has unresolved references:
* `UTC`
* `DEFAULT_TIME_ZONE`
* `Any`
* `DATETIME_RE`